Orange/widgets/settings.py

```diff
@@ -624,8 +624,16 @@ def read_defaults_file(self, settings_file):
         self._migrate_contexts(self.global_contexts)
 
     def _migrate_contexts(self, contexts):
-        for context in contexts:
-            self.widget_class.migrate_context(context, context.values.pop(VERSION_KEY, 0))
+        i = 0
```

Comment:
```suggestion
        migrated = []
        for context in contexts:
            try:
                self.widget_class.migrate_context(
                     context, context.values.pop(VERSION_KEY, 0))
             except IncompatibleContext:
                 pass
         contexts[:] = migrated
```

Reply:
`pass` bi moral biti `migrated.append(context)`? Je to potrebno ali samo stvar okusa? Meni se zdi z brisanjem malo lepše. Če bi selili v `migrated`, bi se mi zdelo bolj prav, da bi `migrate_context` vračal nov kontekst. Ga pa seveda ne bo, ker nima smisla komplicirati.

Reply:
Pri spreminjanju seznama po katerem se iterira me vedno boli glava, ko probavam razumet, kako točno to dela, zato razmišljam o alternativah while-u.

Reply:
```
for i, context in reversed(list(enumerate(contexts))):
    try:
        self.widget_class.migrate_context(
            context, context.values.pop(VERSION_KEY, 0))
    except IncompatibleContext:
        del contexts[i]
```?

Reply:
boljše ideje nimam, če ti ni všeč, lahko ostane tudi tako kot je

Reply:
I'd keep my code. I usually explain its logic that at each step either Muhammad goes to the next mountain (`i += 1`) or the next mountain comes to Muhammad (`del context[i]`). :)

Reply:
Ok, then only tests and documentation are needed :)

Reply:
Sure, I'll see if I know how to write a test for this.

I thought you'd add a warning in the canvas, too?

Reply:
I've added tests and documentations.